### ISSUE TITLE
App pool recycle causes 500 error

### ISSUE THREAD

**Issue by nils-acumen** [dnnsoftware/Dnn.Platform#5381]
## Description of bug
When IIS goes into recycle websites sometimes get stuck in a 500 error.

## Steps to reproduce
1. Change IIS app pool settings from default to
![image](https://user-images.githubusercontent.com/21321363/201168477-22caf3d0-a3fb-4adf-bb0c-f4c38de136e6.png)
2. Wait until the error happens one night.

## Current behavior
Sometimes when IIS tries to recycle the app pool DNN does not come up after that. It just gets stuck in a 500 error.

## Expected behavior
The site comes up without getting stuck in the status 500.

## Error information
Here is an example of the error log:
```
2022-11-11 05:00:32,430 [DHL95V72][Thread:8][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:00:38,296 [DHL95V72][Thread:12][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:00:44,277 [DHL95V72][Thread:27][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:01:26,346 [DHL95V72][Thread:9][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:01:54,109 [DHL95V72][Thread:24][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:02:26,574 [DHL95V72][Thread:8][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:02:54,392 [DHL95V72][Thread:32][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:03:26,413 [DHL95V72][Thread:11][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:03:54,687 [DHL95V72][Thread:28][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:04:25,963 [DHL95V72][Thread:21][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:04:55,566 [DHL95V72][Thread:11][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:04:57,986 [DHL95V72][Thread:9][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:05:06,588 [DHL95V72][Thread:36][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:05:26,704 [DHL95V72][Thread:9][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:05:33,930 [DHL95V72][Thread:42][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:06:26,517 [DHL95V72][Thread:26][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:07:26,705 [DHL95V72][Thread:56][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:08:07,982 [DHL95V72][Thread:26][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:08:26,308 [DHL95V72][Thread:5][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:09:26,122 [DHL95V72][Thread:40][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:10:26,321 [DHL95V72][Thread:25][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:11:26,728 [DHL95V72][Thread:47][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:11:54,628 [DHL95V72][Thread:9][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:12:26,462 [DHL95V72][Thread:54][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:13:25,986 [DHL95V72][Thread:52][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:14:26,685 [DHL95V72][Thread:53][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:15:26,373 [DHL95V72][Thread:5][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:16:26,519 [DHL95V72][Thread:26][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:16:40,071 [DHL95V72][Thread:9][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:16:56,646 [DHL95V72][Thread:54][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:17:26,595 [DHL95V72][Thread:42][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:18:25,972 [DHL95V72][Thread:11][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:19:10,441 [DHL95V72][Thread:37][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:19:26,317 [DHL95V72][Thread:26][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:20:26,588 [DHL95V72][Thread:26][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:21:26,387 [DHL95V72][Thread:24][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:22:25,994 [DHL95V72][Thread:31][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:23:26,699 [DHL95V72][Thread:53][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:24:26,521 [DHL95V72][Thread:37][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:25:26,706 [DHL95V72][Thread:36][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:26:02,469 [DHL95V72][Thread:31][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:26:27,081 [DHL95V72][Thread:9][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:27:25,971 [DHL95V72][Thread:48][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:28:26,534 [DHL95V72][Thread:20][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:29:26,699 [DHL95V72][Thread:26][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:30:26,543 [DHL95V72][Thread:41][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:31:25,955 [DHL95V72][Thread:4][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:32:26,676 [DHL95V72][Thread:29][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:33:26,348 [DHL95V72][Thread:36][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:34:26,480 [DHL95V72][Thread:65][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:35:26,289 [DHL95V72][Thread:73][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:36:26,018 [DHL95V72][Thread:63][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:37:26,304 [DHL95V72][Thread:70][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:38:26,725 [DHL95V72][Thread:48][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:39:26,392 [DHL95V72][Thread:30][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:39:59,442 [DHL95V72][Thread:65][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:40:26,015 [DHL95V72][Thread:30][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:41:26,611 [DHL95V72][Thread:75][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:42:26,431 [DHL95V72][Thread:48][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:43:01,362 [DHL95V72][Thread:59][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:43:26,742 [DHL95V72][Thread:75][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:44:29,609 [DHL95V72][Thread:69][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:45:26,012 [DHL95V72][Thread:8][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:46:26,323 [DHL95V72][Thread:43][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:47:31,472 [DHL95V72][Thread:20][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:48:26,714 [DHL95V72][Thread:8][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:49:25,969 [DHL95V72][Thread:36][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:50:26,794 [DHL95V72][Thread:63][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:51:26,374 [DHL95V72][Thread:72][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:52:26,750 [DHL95V72][Thread:20][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:52:48,759 [DHL95V72][Thread:78][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:53:26,702 [DHL95V72][Thread:70][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:54:08,710 [DHL95V72][Thread:43][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:54:25,961 [DHL95V72][Thread:64][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:55:26,348 [DHL95V72][Thread:44][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:56:26,713 [DHL95V72][Thread:65][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:57:26,406 [DHL95V72][Thread:14][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:58:26,017 [DHL95V72][Thread:73][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 05:59:26,744 [DHL95V72][Thread:60][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 06:00:26,375 [DHL95V72][Thread:60][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 06:01:26,706 [DHL95V72][Thread:75][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 06:02:26,658 [DHL95V72][Thread:68][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
2022-11-11 06:03:25,972 [DHL95V72][Thread:86][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
```

## Additional context
- We're using Pingdom.com to monitor the status of our websites.
- This started happening only after we upgraded our SQL env. (SQL2012 to SQL2019, the server also changed from a physical one to a VM)
- Manually restarting the app pool resolves the issue
- It looks like its [this issue reported here](https://www.dnnsoftware.com/answers/error-500-after-windows-server-reboot). However we're not restarting the server. It's triggered on the app pool recycle as stated above.

## Affected version

* [ ] 10.00.00 alpha build
* [ ] 09.11.00 release candidate
* [ ] 09.10.02 release candidate
* [x] 09.10.01 latest supported release
* [x] 09.01.01

## Affected browser

* [x] Chrome
* [x] Firefox
* [x] Safari
* [x] Internet Explorer 11
* [x] Microsoft Edge (Classic)
* [x] Microsoft Edge Chromium
This is not related to the browser.

**Comment by sleupold** [comment]
this doesn't sound like an issue of the website due an app pool recycle, but a problem after a server restart, when the SQL server neds longer to come up than the IIS server.

**Comment by nils-acumen** [comment]
If you mean the link, yes I would agree. 

The issue that we're experiencing is somewhat different since there was no server restart involved whatsoever. When you look at the logs at 5am it would have recycled the app pool and then it just didn't manage to spin DNN back up. I'm just trying to understand why that is.

I mean I could probably create a ps script that checks the status of the site and if it returns 500 just recycle the app pool. The thing that puts me off is that this never used to happen.

**Comment by mitchelsellers** [comment]
This is something that can happen due to a number of errors, and other configurations.

* Rapid Fail Production - If the app is forced to recycle and encounters a number of errors at first, it can trigger Rapid Fail Protection which will stop it from starting
* Modules with Incorrect SQL Management - I've seen issues with SQL reconnecting after repeated app restarts due t improper connection management by third-party modules.  Essentially using up too many threads for connections and blocking communications.
* Generic Nature of Scheduled Recycles - Traditionally, arbitrary recycles are very dangerous, as you have no real expectation of load behaviors or patterns.  Every 5 hours could trigger a restart during heavy load, which could trigger one of the above issues.  I do NOT recommend arbitrarily scheduled recycles, if you MUST recycle on a schedule pick a fixed, low-traffic time.

Lastly, what version of DNN?

**Comment by nils-acumen** [comment]
Thanks for the answer.

Yes this instance is running on 9.01.01 will push for an upgrade. I am aware of the security issues. However, I had this happening to other sites as well. This might actually only affect these older versions.

Can I as how you typically configure the recycling?

**Comment by mitchelsellers** [comment]
I would for sure upgrade before diving too far in. 

I do not adjust the defaults. So no scheduled recycle.

**Comment by valadas** [comment]
My recommendation would be to start by upgrading to the latest version because:
1. Security
2. Maybe it's no longer an issue, reporting bugs about older versions is not very productive if we can't make it happen with the current code. A lot of water went under the bridges from 9.1.1 to 9.11.0.
3. 9.11.0 had massive performance improvements and it may make it so this is no longer an issue (uncertain but worth trying)

I'll close this issue for now, if you can still make it happen after upgrading feel free to comment or re-open a new issue.
